### ISSUE TITLE
ban-null-checks-against-arrays

### ISSUE THREAD

**Issue by jpike88** [eslint/eslint#14563]
**Please describe what the rule should do:**

Sometimes an array will be instantiated and accidentally null checked instead of 0 length checked

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)

**Provide 2-3 code examples that this rule will warn about:**

```js
const x = [];

// BAN THIS
if(!x){}
```

**Are you willing to submit a pull request to implement this rule?**
No

**Comment by nzakas** [comment]
Thanks for the suggestion. This type of rule requires type checking, which we don’t support. You can probably make such a rule yourself if you want to keep track of arrays but this isn’t a good fit as a core rule.
